kiwipy/communications.py

```diff
@@ -129,26 +137,31 @@ def remove_rpc_subscriber(self, identifier):
         except KeyError:
             raise ValueError("Unknown subscriber '{}'".format(identifier))
 
-    def add_task_subscriber(self, subscriber):
+    def add_task_subscriber(self, subscriber, identifier=None):
         """
         Register a task subscriber
 
         :param subscriber: The task callback function
+        :param identifier: the subscriber identifier
         """
         self._ensure_open()
-        self._task_subscribers.append(subscriber)
+        identifier = identifier or shortuuid.uuid()
+        if identifier in self._rpc_subscribers:
+            raise exceptions.DuplicateSubscriberIdentifier("RPC identifier '{}'".format(identifier))
+        self._task_subscribers[identifier] = subscriber
+        return identifier
 
-    def remove_task_subscriber(self, subscriber):
+    def remove_task_subscriber(self, identifier):
         """
         Remove a task subscriber
 
-        :param subscriber: The task callback function
+        :param identifier: The task subscriber identifier
```

Comment:
Maybe add that it raises `ValueError` if `identifier` is not existing task subscriber

Reply:
Done.